far/dialog.cpp

```diff
@@ -4229,12 +4229,12 @@ void Dialog::Process()
 
 intptr_t Dialog::CloseDialog()
 {
-	GetDialogObjectsData();
-
 	const auto result = DlgProc(DN_CLOSE, m_ExitCode, nullptr);
 	if (!result)
 		return 0;
 
+	GetDialogObjectsData();
```

Comment:
This function does some other stuff, e.g. propagates some data from EditPtr and ListPtr back to dialog items.
DN_CLOSE handlers might expect that, so we can't just move the whole thing.

Could you please try the attached? (and update the PR if it works as expected)
[946.diff.txt](https://github.com/user-attachments/files/20040334/946.diff.txt)

Reply:
Works like a charm